internal/pkg/application/handler.go

```diff
@@ -259,6 +260,8 @@ func (cs *contextSource) RetrieveEntity(entityID string, req ngsi.Request) (ngsi
 			)
 		}
 
+		fiwareDevice.Location = geojson.CreateGeoJSONPropertyFromWGS84(device.Longitude, device.Latitude)
```

Comment:
Make this conditional on if the device actually has a location (ie: location differs from (0.0, 0.0)).